backend/app/verification.py

```diff
@@ -104,23 +105,34 @@ def _get_domain_name(appid: str) -> str:
         return ".".join(reversed(appid.split(".")[0:2]))
 
 
-def _check_app_id_error(appid: str) -> str:
-    """
-    Returns an ErrorDetail if the app ID is invalid, or None if it is valid.
-    """
-    if len(appid.split(".")) < 3:
-        return ErrorDetail.MALFORMED_APP_ID
-    elif not re.match("[_\w\.]+$", appid):
-        return ErrorDetail.MALFORMED_APP_ID
+def is_valid_app_id(appid: str) -> bool:
+    return len(appid.split(".")) >= 3 and re.match("[_\w\.]+$", appid)
+
+
+def is_github_app(appid: str) -> bool:
+    """Determines whether the app is a repo in github.com/flathub."""
 
     try:
         r = requests.get(
             f"https://api.github.com/repos/flathub/{urllib.parse.quote(appid, safe='')}"
         )
-        if r.status_code != 200:
-            return ErrorDetail.REPO_DOES_NOT_EXIST
+        return r.status_code == 200
     except:
-        return ErrorDetail.ERROR_CONNECTING_TO_GITHUB
+        raise HTTPException(
+            status_code=500, detail=ErrorDetail.ERROR_CONNECTING_TO_GITHUB
+        )
+
+
+def _check_app_id_error(appid: str) -> ErrorDetail:
+    """
+    Returns an ErrorDetail if the app ID is invalid, or None if it is valid.
+    """
+
+    if not is_valid_app_id(appid):
+        return ErrorDetail.MALFORMED_APP_ID
+
+    if not is_github_app(appid):
+        return ErrorDetail.REPO_DOES_NOT_EXIST
```

Comment:
Wondering if we might want to make this error github specific, so that it's clear we men the github repo.